### PR TITLE
Run tpcds benchmarks for accelerators

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

- This PR adds TPCDS benchmark tests for Arrow, DuckDB, Postgres, and SQLite accelerator, and updates the benchmark workflow to run these accelerator TPCDS tests.

- This PR also includes indexes to speed up Postgres TPCDS benchmark. Certain queries that are slow due to costly rewritten plans are commented out, and the relevant issue is created: https://github.com/spiceai/spiceai/issues/2939

- Note that SQLite file mode acceleration would fail due to slow queries. It's still under discussion on how to move forward with the SQLite TPCDS benchmark test. Changes to make SQLite TPCDS benchmark test run will be addressed in incremental PRs after this one.

## 🔨 Related Issues

- https://github.com/spiceai/spiceai/issues/2827
- https://github.com/spiceai/spiceai/issues/2929
- https://github.com/spiceai/spiceai/issues/2828
- https://github.com/spiceai/spiceai/issues/2830

### CODE REVIEW

**Comment by sgrebnov** [comment]
@Sevenannn  - if this will require some extra work, can we please just add TPC-DS test queries first, I'd like to modify some of them so they are Postgres dialect compliant and start tracking TPC-DS notes/quirks.

**Comment by Sevenannn** [comment]
@sgrebnov Sure I can do that, I'll ping you when the PR is done

File: crates/runtime/benches/bench_s3/mod.rs [hunk 1]
```diff
@@ -28,8 +28,19 @@ pub(crate) async fn run(
     benchmark_results: &mut BenchmarkResultsBuilder,
     engine: Option<String>,
     mode: Option<Mode>,
+    bench_name: &str,
 ) -> Result<(), String> {
-    let test_queries = get_test_queries();
+    let test_queries = match bench_name {
+        "tpch" => get_test_queries(),
+        "tpcds" => {
+            if engine.clone().unwrap_or_default().as_str() == "postgres" {
```
Comment (peasee) [inline crates/runtime/benches/bench_s3/mod.rs:37]: might be worth adding a comment here that postgres has some  query modifications to make TPC-DS pass

File: crates/runtime/benches/bench_s3/mod.rs [hunk 2]
```diff
@@ -63,28 +74,126 @@ pub(crate) async fn run(
...
+            ))
+            .with_dataset(make_dataset(
+                "spiceai-public-datasets/tpcds/web_page/",
+                "web_page",
+            ))
+            .with_dataset(make_dataset(
+                "spiceai-public-datasets/tpcds/web_site/",
+                "web_site",
+            )),
+        _ => unreachable!(),
```
Comment (peasee) [inline crates/runtime/benches/bench_s3/mod.rs]:
```suggestion
        _ => panic!("Only tpcds or tpch benchmark suites are supported"),
```

add a message to this unreachable and make it a `panic!()` - it's not unreachable if someone just passes `-b not_a_real_benchmark` :smile:

File: crates/runtime/benches/bench_s3/mod.rs [hunk 3]
```diff
@@ -151,3 +260,215 @@ fn get_test_queries() -> Vec<(&'static str, &'static str)> {
...
+        ("tpcds_q97", include_str!("../queries/tpcds/q97.sql")),
+        ("tpcds_q98", include_str!("../queries/tpcds/q98.sql")),
+        ("tpcds_q99", include_str!("../queries/tpcds/q99.sql")),
+    ]
+}
+
+#[allow(clippy::too_many_lines)]
+fn get_postgres_tpcds_test_queries() -> Vec<(&'static str, &'static str)> {
+    vec![
+        // Query 1, 30, 64, 81 commented out due to rewritten query's expensive plan in Postgres
```
Comment (peasee) [inline crates/runtime/benches/bench_s3/mod.rs:373]: nice, I see the comment here now. Let's incude a shorter blurb version next to the actual `if {}`

File: crates/runtime/benches/bench.rs [hunk 4]
```diff
@@ -169,18 +187,27 @@ async fn main() -> Result<(), String> {
 async fn run_connector_bench(
     connector: &str,
     upload_results_dataset: &Option<String>,
+    bench_name: &str,
 ) -> Result<(), String> {
+    // TODO: Implement and enable connector TPCDS bench if it's required
+    if bench_name == "tpcds" {
+        return Err(
+            "TPCDS Benchmark not implemented for data connectors, please implement if needed"
```
Comment (phillipleblanc) [inline crates/runtime/benches/bench.rs]:
```suggestion
            "TPCDS Benchmark not implemented for data connectors"
```

**Review by phillipleblanc — DISMISSED** [review]

**Review by phillipleblanc — DISMISSED** [review]

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**